renku_notebooks/api/notebooks.py

```diff
@@ -295,11 +307,140 @@ def launch_notebook(
     manifest = server.start()
 
     current_app.logger.debug(f"Server {server.server_name} has been started")
-    for autosave in user.get_autosaves(server.gl_project.path_with_namespace):
-        autosave.cleanup(server.commit_sha)
+
     return NotebookResponse().dump(UserServerManifest(manifest)), 201
 
 
+@bp.route("servers/<server_name>", methods=["PATCH"])
+@use_args(
+    {"server_name": fields.Str(required=True)}, location="view_args", as_kwargs=True
+)
+@use_args(PatchServerRequest(), location="json", as_kwargs=True)
+@authenticated
+def patch_server(user, server_name, state):
+    """
+    Patch a user server by name based on the query param.
+
+    ---
+    patch:
+      description: Patch a running server by name.
+      requestBody:
+        content:
+          application/json:
+            schema: PatchServerRequest
+      parameters:
+        - in: path
+          schema:
+            type: string
+          name: server_name
+          required: true
+          description: The name of the server that should be patched.
+      responses:
+        204:
+          description: The server was patched successfully.
+          content:
+            application/json:
+              schema: NotebookResponse
+        400:
+          description: Invalid json argument value.
+          content:
+            application/json:
+              schema: ErrorResponse
+        404:
+          description: The server cannot be found.
+          content:
+            application/json:
+              schema: ErrorResponse
+        500:
+          description: The server exists but could not be successfully hibernated.
+          content:
+            application/json:
+              schema: ErrorResponse
+      tags:
+        - servers
+    """
+    if not config.sessions.storage.pvs_enabled:
+        raise PVDisabledError()
+
+    if isinstance(user, AnonymousUser):
+        raise AnonymousUserPatchError()
+
+    logging.warning(f"State is {state}")
+    logging.warning(
+        f"H = {PatchServerStatusEnum.Hibernated.value}, R = {PatchServerStatusEnum.Running.value}"
+    )
+
+    server = config.k8s.client.get_server(server_name, user.safe_username)
+
+    if state == PatchServerStatusEnum.Hibernated.value:
+        # NOTE: Do nothing if server is already hibernated
+        if server and server.get("spec", {}).get("jupyterServer", {}).get(
+            "hibernated", False
+        ):
+            logging.warning(f"Server {server_name} is already hibernated.")
+
+            return NotebookResponse().dump(UserServerManifest(server)), 204
+
+        status = get_status(server_name=server_name, access_token=user.access_token)
```

Comment:
This will return None if the sidecar is not available. We should handle that. The return type of get status is wrong. Sorry we do not have mypy here.